src/UnitfulRecipes.jl

```diff
@@ -27,42 +29,41 @@ function fixaxis!(attr, x, axisletter)
     axislims = Symbol(axisletter, :lims)   # xlims, ylims, zlims
     axisticks = Symbol(axisletter, :ticks) # xticks, yticks, zticks
     err = Symbol(axisletter, :error)       # xerror, yerror, zerror
-    axisunit = Symbol(axisletter, :unit)   # xunit, yunit, zunit
-    axis = Symbol(axisletter, :axis)       # xaxis, yaxis, zaxis
-    # Get the unit
-    u = pop!(attr, axisunit, unit(eltype(x)))
-    # If the subplot already exists with data, get its unit
-    sp = get(attr, :subplot, 1)
-    if sp ≤ length(attr[:plot_object]) && attr[:plot_object].n > 0
-        label = attr[:plot_object][sp][axis][:guide]
-        if label isa UnitfulString
-            u = label.unit
-        end
-        # If label was not given as an argument, reuse
-        get!(attr, axislabel, label)
-    end
+
+    u = fixlabel!(attr, axisletter, unit(first(x)))
     # Fix the attributes: labels, lims, ticks, marker/line stuff, etc.
     append_unit_if_needed!(attr, axislabel, u)
     ustripattribute!(attr, axislims, u)
     ustripattribute!(attr, axisticks, u)
     ustripattribute!(attr, err, u)
-    fixmarkercolor!(attr)
+    fixcolors!(attr)
```

Comment:
Need to run a single pass over all three color arguments, to combine into one axis.